// qaseio/dist/src/qaseio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QaseApi = void 0;
var axios_1 = __importDefault(require("axios"));
var attachments_1 = require("./services/attachments");
var cases_1 = require("./services/cases");
var customFields_1 = require("./services/customFields");
var defects_1 = require("./services/defects");
var milestones_1 = require("./services/milestones");
var plans_1 = require("./services/plans");
var projects_1 = require("./services/projects");
var results_1 = require("./services/results");
var runs_1 = require("./services/runs");
var sharedSteps_1 = require("./services/sharedSteps");
var suites_1 = require("./services/suites");
var users_1 = require("./services/users");
var QaseApi = /** @class */ (function () {
    function QaseApi(apiToken) {
        var config = {
            headers: {
                Token: apiToken,
            },
            baseURL: 'https://api.qase.io/v1',
        };
        this.api = axios_1.default.create(config);
        this.projects = new projects_1.Projects(this.api);
        this.cases = new cases_1.Cases(this.api);
        this.results = new results_1.Results(this.api);
        this.runs = new runs_1.Runs(this.api);
        this.attachments = new attachments_1.Attachments(this.api);
        this.plans = new plans_1.Plans(this.api);
        this.suites = new suites_1.Suites(this.api);
        this.milestones = new milestones_1.Milestones(this.api);
        this.sharedSteps = new sharedSteps_1.SharedSteps(this.api);
        this.defects = new defects_1.Defects(this.api);
        this.customFields = new customFields_1.CustomFields(this.api);
        this.users = new users_1.Users(this.api);
    }
    return QaseApi;
}());
exports.QaseApi = QaseApi;
