// qaseio/dist/src/models/results.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResultUpdate = exports.ResultCreate = exports.ResultStepCreate = exports.ResultFilters = exports.ResultStatus = void 0;
var ResultStatus;
(function (ResultStatus) {
    ResultStatus["IN_PROGRESS"] = "in_progress";
    ResultStatus["PASSED"] = "passed";
    ResultStatus["FAILED"] = "failed";
    ResultStatus["SKIPPED"] = "skipped";
    ResultStatus["BLOCKED"] = "blocked";
})(ResultStatus = exports.ResultStatus || (exports.ResultStatus = {}));
var ResultFilters = /** @class */ (function () {
    function ResultFilters(filter) {
        this.filter = filter;
    }
    return ResultFilters;
}());
exports.ResultFilters = ResultFilters;
var ResultStepCreate = /** @class */ (function () {
    function ResultStepCreate(position, status, attachments, comment) {
        if (attachments === void 0) { attachments = []; }
        this.position = position;
        this.status = status;
        this.attachments = attachments;
        this.comment = comment;
    }
    return ResultStepCreate;
}());
exports.ResultStepCreate = ResultStepCreate;
var ResultCreate = /** @class */ (function () {
    function ResultCreate(case_id, status, args) {
        this.case_id = case_id;
        this.status = status;
        Object.assign(this, args);
    }
    return ResultCreate;
}());
exports.ResultCreate = ResultCreate;
var ResultUpdate = /** @class */ (function () {
    function ResultUpdate(args) {
        Object.assign(this, args);
    }
    return ResultUpdate;
}());
exports.ResultUpdate = ResultUpdate;
