// qaseio/dist/src/services/sharedSteps.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedSteps = void 0;
var models_1 = require("../models");
var _1 = require(".");
var SharedSteps = /** @class */ (function (_super) {
    __extends(SharedSteps, _super);
    function SharedSteps() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SharedSteps.prototype.getAll = function (code, _a) {
        var limit = _a.limit, offset = _a.offset, filters = _a.filters;
        var filterParams = new models_1.Filter(filters === null || filters === void 0 ? void 0 : filters.filter).filter();
        return this.api
            .get("/shared_step/" + code, { params: __assign({ limit: limit, offset: offset }, filterParams) })
            .then(this.validateResponse());
    };
    SharedSteps.prototype.get = function (code, sharedStepHash) {
        return this.api
            .get("/shared_step/" + code + "/" + sharedStepHash)
            .then(this.validateResponse());
    };
    SharedSteps.prototype.exists = function (code, sharedStepHash) {
        return this.get(code, sharedStepHash)
            .then(function (resp) { return Boolean(resp.data.hash); })
            .catch(function () { return false; });
    };
    SharedSteps.prototype.create = function (code, data) {
        return this.api
            .post("/shared_step/" + code, data)
            .then(this.validateResponse());
    };
    SharedSteps.prototype.update = function (code, sharedStepHash, data) {
        return this.api
            .patch("/shared_step/" + code + "/" + sharedStepHash, data)
            .then(this.validateResponse());
    };
    SharedSteps.prototype.delete = function (code, sharedStepHash) {
        return this.api
            .delete("/shared_step/" + code + "/" + sharedStepHash)
            .then(this.validateResponse());
    };
    return SharedSteps;
}(_1.BaseService));
exports.SharedSteps = SharedSteps;
