// qaseio/dist/src/services/suites.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Suites = void 0;
var models_1 = require("../models");
var _1 = require(".");
var Suites = /** @class */ (function (_super) {
    __extends(Suites, _super);
    function Suites() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Suites.prototype.getAll = function (code, _a) {
        var limit = _a.limit, offset = _a.offset, filters = _a.filters;
        var filterParams = new models_1.Filter(filters === null || filters === void 0 ? void 0 : filters.filter).filter();
        return this.api
            .get("/suite/" + code, { params: __assign({ limit: limit, offset: offset }, filterParams) })
            .then(this.validateResponse());
    };
    Suites.prototype.get = function (code, suiteId) {
        return this.api
            .get("/suite/" + code + "/" + suiteId)
            .then(this.validateResponse());
    };
    Suites.prototype.exists = function (code, suiteId) {
        return this.get(code, suiteId)
            .then(function (resp) { return Boolean(resp.data.id); })
            .catch(function () { return false; });
    };
    Suites.prototype.create = function (code, data) {
        return this.api
            .post("/suite/" + code, data)
            .then(this.validateResponse());
    };
    Suites.prototype.update = function (code, suiteId, data) {
        return this.api
            .patch("/suite/" + code + "/" + suiteId, data)
            .then(this.validateResponse());
    };
    Suites.prototype.delete = function (code, suiteId) {
        return this.api
            .delete("/suite/" + code + "/" + suiteId)
            .then(this.validateResponse());
    };
    return Suites;
}(_1.BaseService));
exports.Suites = Suites;
